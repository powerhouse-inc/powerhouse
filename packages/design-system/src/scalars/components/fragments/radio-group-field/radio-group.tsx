import React, { useId } from "react";
import * as RadioGroupPrimitive from "@radix-ui/react-radio-group";

export interface CustomizableRadioGroupProps
  extends React.ComponentPropsWithoutRef<typeof RadioGroupPrimitive.Root> {
  children: React.ReactNode;
  className?: string;
  id?: string;
  name?: string;
}

export const RadioGroup = React.forwardRef<
  React.ElementRef<typeof RadioGroupPrimitive.Root>,
  CustomizableRadioGroupProps
>(({ children, className, id: propId, name, ...props }, ref) => {
  const prefix = useId();
  const id = propId ?? `${prefix}-radio-group`;

  return (
    <RadioGroupPrimitive.Root
      className={className}
      id={id}
      name={name}
      {...props}
      ref={ref}
    >
      {children}
    </RadioGroupPrimitive.Root>
  );
});

RadioGroup.displayName = "RadioGroup";
