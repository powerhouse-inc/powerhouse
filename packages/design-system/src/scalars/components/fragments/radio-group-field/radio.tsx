import React, { useId } from "react";
import * as RadioGroupPrimitive from "@radix-ui/react-radio-group";
import { FormLabel } from "@/scalars/components/fragments/form-label";
import { cn } from "@/scalars/lib/utils";

export interface RadioProps
  extends React.ComponentPropsWithoutRef<typeof RadioGroupPrimitive.Item> {
  className?: string;
  description?: string;
  disabled?: boolean;
  hasError?: boolean;
  id?: string;
  label: string;
  value: string;
}

export const Radio = React.forwardRef<
  React.ElementRef<typeof RadioGroupPrimitive.Item>,
  RadioProps
>(
  (
    {
      className,
      description,
      disabled = false,
      hasError = false,
      id: propId,
      label,
      value,
      ...props
    },
    ref,
  ) => {
    const prefix = useId();
    const id = propId ?? `${prefix}-radio`;

    return (
      <>
        <RadioGroupPrimitive.Item
          aria-disabled={disabled}
          aria-invalid={hasError}
          className={cn(
            "group",
            "aspect-square size-4 rounded-full border-2 border-gray-800",
            "dark:border-gray-400",
            "focus:outline-none focus:ring-2 focus:ring-gray-300/40 focus:ring-offset-0",
            "dark:focus:ring-gray-500/40",
            "hover:border-gray-900",
            "dark:hover:border-gray-50",
            "focus:hover:ring-gray-400/40",
            "dark:focus:hover:ring-gray-400/40",
            disabled && [
              "cursor-not-allowed border-gray-600 opacity-70 hover:border-gray-600",
              "dark:border-gray-600 dark:hover:border-gray-600",
            ],
            hasError && [
              "border-red-700 hover:border-red-900 focus:ring-transparent focus:hover:ring-transparent",
              "dark:border-red-700 dark:hover:border-red-900 dark:focus:ring-transparent dark:focus:hover:ring-transparent",
            ],
            className,
          )}
          disabled={disabled}
          id={id}
          value={value}
          {...props}
          ref={ref}
        >
          <RadioGroupPrimitive.Indicator
            className={cn(
              "relative flex size-full items-center justify-center",
              "after:block after:size-2 after:rounded-full after:bg-gray-800 after:content-['']",
              "dark:after:bg-gray-400",
              "group-hover:after:bg-gray-900",
              "dark:group-hover:after:bg-gray-50",
              disabled && [
                "after:bg-gray-600 after:hover:bg-gray-600",
                "dark:after:bg-gray-600 dark:after:hover:bg-gray-600",
              ],
            )}
          />
        </RadioGroupPrimitive.Item>
        <FormLabel
          description={description}
          disabled={disabled}
          hasError={hasError}
          htmlFor={id}
          required={props.required}
        >
          {label}
        </FormLabel>
      </>
    );
  },
);
