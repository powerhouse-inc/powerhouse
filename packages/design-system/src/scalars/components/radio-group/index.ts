export * from "./radio";
export * from "./radio-group";
export * from "./radio-group-field";
