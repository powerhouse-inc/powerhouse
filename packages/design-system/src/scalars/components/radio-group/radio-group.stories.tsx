import React from "react";
import type { Meta, StoryObj } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { RadioGroupField } from "./radio-group-field";

const meta: Meta<typeof RadioGroupField> = {
  argTypes: {
    className: {
      control: "text",
      description: "Additional CSS classes to apply to the Radio Group",
    },
    defaultValue: {
      control: "text",
      description: "Default selected value for the Radio Group",
    },
    description: {
      control: "text",
      description: "Description for the Radio Group",
    },
    errors: {
      control: "object",
      description: "Array of error messages to display",
      table: { defaultValue: { summary: "[]" } },
    },
    label: {
      control: "text",
      description: "Label for the Radio Group",
    },
    onChange: {
      action: "onChange",
      description: "Callback fired when the value changes",
    },
    radioOptions: {
      control: "object",
      description:
        "Array of radio options with label, value and optional description",
      table: { defaultValue: { summary: "[]" } },
    },
    required: {
      control: "boolean",
      description:
        "Indicates if selecting an option in the Radio Group is required",
      table: { defaultValue: { summary: "false" } },
    },
    value: {
      control: "text",
      description: "Currently selected value in the Radio Group",
    },
  },
  component: RadioGroupField,
  parameters: {
    controls: { sort: "requiredFirst" },
  },
  tags: ["autodocs"],
  title: "Document Engineering/Simple Components/Radio Group/Radio Group",
};

export default meta;

type Story = StoryObj<typeof RadioGroupField>;

export const WithoutLabel: Story = {
  args: {
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2" },
      { label: "Option 3", value: "3" },
    ],
  },
};

export const WithLabel: Story = {
  args: {
    label: "Radio Group with label",
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2" },
      { label: "Option 3", value: "3" },
    ],
  },
};

export const WithLabelAndDescription: Story = {
  args: {
    description: "This is a helpful description for the Radio Group",
    label: "Radio Group with label and description",
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2" },
      { label: "Option 3", value: "3" },
    ],
  },
};

export const WithErrors: Story = {
  args: {
    errors: ["Error 1", "Error 2"],
    label: "Radio Group with errors",
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2" },
      { label: "Option 3", value: "3" },
    ],
  },
};

export const Required: Story = {
  args: {
    label: "Required Radio Group",
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2" },
      { label: "Option 3", value: "3" },
    ],
    required: true,
  },
};

export const WithOptionSelectedByDefault: Story = {
  args: {
    defaultValue: "2",
    label: "Radio Group with option selected by default",
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2" },
      { label: "Option 3", value: "3" },
    ],
  },
};

const ControlledRadioGroup = () => {
  const [value, setValue] = React.useState("1");
  const handleChange = (newValue: string) => {
    setValue(newValue);
    action("onChange")(newValue);
  };

  return (
    <RadioGroupField
      label="Controlled Radio Group"
      onChange={handleChange}
      radioOptions={[
        { label: "Option 1", value: "1" },
        { label: "Option 2", value: "2" },
        { label: "Option 3", value: "3" },
      ]}
      value={value}
    />
  );
};

export const Controlled: Story = {
  render: () => <ControlledRadioGroup />,
  parameters: {
    docs: {
      source: {
        code: `
const ControlledRadioGroup = () => {
  const [value, setValue] = React.useState("1");
  const handleChange = (newValue: string) => {
    setValue(newValue);
  };

  return (
    <RadioGroupField
      label="Controlled Radio Group"
      onChange={handleChange}
      radioOptions={[
        { label: "Option 1", value: "1" },
        { label: "Option 2", value: "2" },
        { label: "Option 3", value: "3" },
      ]}
      value={value}
    />
  );
};
        `,
      },
    },
  },
};

export const WithDescriptionInOptions: Story = {
  args: {
    label: "Radio Group with description in the options",
    radioOptions: [
      {
        description: "Description for option 1",
        label: "Option 1",
        value: "1",
      },
      {
        description: "Description for option 2",
        label: "Option 2",
        value: "2",
      },
    ],
  },
};

export const WithDisabledOptions: Story = {
  args: {
    label: "Radio Group with disabled options",
    radioOptions: [
      { label: "Option 1", value: "1" },
      { label: "Option 2 (Disabled)", value: "2", disabled: true },
      { label: "Option 3", value: "3" },
      {
        label: "Option 4 (Disabled with description)",
        value: "4",
        disabled: true,
        description: "This option is disabled and has a description",
      },
    ],
  },
};
