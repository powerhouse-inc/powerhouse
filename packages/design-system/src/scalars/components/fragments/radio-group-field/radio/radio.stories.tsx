import type { Meta, StoryObj } from "@storybook/react";
import { Radio } from "./radio";
import { RadioGroup } from "../radio-group";

const meta: Meta<typeof Radio> = {
  argTypes: {
    checked: {
      control: "boolean",
      description: "Indicates if the Radio is checked",
    },
    className: {
      control: "text",
      description: "Additional CSS classes to apply to the Radio",
    },
    description: {
      control: "text",
      description: "Description for the Radio",
    },
    disabled: {
      control: "boolean",
      description: "Indicates if the Radio is disabled",
      table: { defaultValue: { summary: "false" } },
    },
    label: {
      control: "text",
      description: "Label for the Radio",
      table: { defaultValue: { summary: "" } },
    },
    value: {
      control: "text",
      description: "Value of the Radio input",
      table: { defaultValue: { summary: "" } },
    },
  },
  component: Radio,
  decorators: [
    (Story) => (
      <RadioGroup name="radio-group">
        <div className="flex items-center space-x-2.5" role="presentation">
          <Story />
        </div>
      </RadioGroup>
    ),
  ],
  title: "Document Engineering/Fragments/Radio",
};

export default meta;

type Story = StoryObj<typeof Radio>;

export const Default: Story = {
  args: {
    label: "Default",
    value: "default",
  },
};

export const DefaultChecked: Story = {
  args: {
    checked: true,
    label: "Default Checked",
    value: "default-checked",
  },
};

export const DefaultWithDescription: Story = {
  args: {
    description: "This is a default Radio with description",
    label: "Default with description",
    value: "default-with-description",
  },
};

export const DefaultWithDescriptionChecked: Story = {
  args: {
    checked: true,
    description: "This is a default Radio with description and checked",
    label: "Default with description and checked",
    value: "default-with-description-checked",
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    label: "Disabled",
    value: "disabled",
  },
};

export const DisabledChecked: Story = {
  args: {
    checked: true,
    disabled: true,
    label: "Disabled checked",
    value: "disabled-checked",
  },
};

export const DisabledWithDescription: Story = {
  args: {
    description: "This is a disabled Radio with description",
    disabled: true,
    label: "Disabled with description",
    value: "disabled-with-description",
  },
};

export const DisabledWithDescriptionChecked: Story = {
  args: {
    checked: true,
    description: "This is a disabled Radio with description and checked",
    disabled: true,
    label: "Disabled with description and checked",
    value: "disabled-with-description-checked",
  },
};

export const Focus: Story = {
  args: {
    label: "Focus",
    value: "focus",
  },
  parameters: {
    pseudo: { focus: true },
  },
};

export const FocusChecked: Story = {
  args: {
    checked: true,
    label: "Focus Checked",
    value: "focus-checked",
  },
  parameters: {
    pseudo: { focus: true },
  },
};

export const FocusWithDescription: Story = {
  args: {
    description: "This is a focused Radio with description",
    label: "Focus with description",
    value: "focus-with-description",
  },
  parameters: {
    pseudo: { focus: true },
  },
};

export const FocusWithDescriptionChecked: Story = {
  args: {
    checked: true,
    description: "This is a focused Radio with description and checked",
    label: "Focus with description and checked",
    value: "focus-with-description-checked",
  },
  parameters: {
    pseudo: { focus: true },
  },
};

export const Hover: Story = {
  args: {
    label: "Hover",
    value: "hover",
  },
  parameters: {
    pseudo: { hover: true },
  },
};

export const HoverChecked: Story = {
  args: {
    checked: true,
    label: "Hover Checked",
    value: "hover-checked",
  },
  parameters: {
    pseudo: { hover: true },
  },
};

export const HoverWithDescription: Story = {
  args: {
    description: "This is a hovered Radio with description",
    label: "Hover with description",
    value: "hover-with-description",
  },
  parameters: {
    pseudo: { hover: true },
  },
};

export const HoverWithDescriptionChecked: Story = {
  args: {
    checked: true,
    description: "This is a hovered Radio with description and checked",
    label: "Hover with description and checked",
    value: "hover-with-description-checked",
  },
  parameters: {
    pseudo: { hover: true },
  },
};
