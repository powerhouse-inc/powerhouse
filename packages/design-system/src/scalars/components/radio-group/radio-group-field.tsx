import React, { useId } from "react";
import { FormLabel } from "@/scalars/components/form-label";
import { FormMessageList } from "@/scalars/components/form-message";
import { Radio } from "./radio";
import { RadioGroup } from "./radio-group";

export interface RadioOption {
  label: string;
  value: string;
  description?: string;
  disabled?: boolean;
}

export interface RadioGroupFieldProps {
  className?: string;
  defaultValue?: string;
  description?: string;
  errors?: string[];
  id?: string;
  label?: string;
  name?: string;
  onChange?: (value: string) => void;
  radioOptions: RadioOption[];
  required?: boolean;
  value?: string;
}

export const RadioGroupField: React.FC<RadioGroupFieldProps> = ({
  className,
  defaultValue,
  description,
  errors = [],
  id,
  label,
  name,
  onChange,
  radioOptions = [],
  required = false,
  value,
}) => {
  const prefix = useId();
  const hasLabel = label !== undefined;
  const hasError = errors.length > 0;

  return (
    <RadioGroup
      aria-invalid={hasError}
      aria-required={required}
      aria-label={!hasLabel ? "Radio group" : undefined}
      className={className}
      defaultValue={defaultValue}
      id={id}
      name={name}
      onValueChange={onChange}
      value={value}
    >
      {hasLabel && (
        <FormLabel
          description={description}
          hasError={hasError}
          htmlFor={id}
          required={required}
        >
          {label}
        </FormLabel>
      )}
      {radioOptions.map((option, index) => (
        <div
          key={`${prefix}-radio-${index}-${option.value}`}
          className="flex items-center gap-2.5"
          role="presentation"
        >
          <Radio
            id={`${prefix}-radio-${index}-${option.value}`}
            label={option.label}
            value={option.value}
            description={option.description}
            disabled={option.disabled}
            hasError={hasError}
          />
        </div>
      ))}
      {hasError && <FormMessageList messages={errors} type="error" />}
    </RadioGroup>
  );
};
