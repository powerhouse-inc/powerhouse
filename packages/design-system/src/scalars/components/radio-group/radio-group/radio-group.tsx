import React, { useId } from "react";
import * as RadioGroupPrimitive from "@radix-ui/react-radio-group";
import { FormLabel } from "@/scalars/components/form-label";
import { cn } from "@/scalars/lib/utils";

export interface RadioGroupProps
  extends React.ComponentPropsWithoutRef<typeof RadioGroupPrimitive.Root> {
  "aria-label"?: string;
  children: React.ReactElement;
  className?: string;
  defaultValue?: string;
  description?: string;
  hasError?: boolean;
  id?: string;
  label?: string;
  name?: string;
  onValueChange?: (value: string) => void;
  required?: boolean;
  value?: string;
}

export const RadioGroup = React.forwardRef<
  React.ElementRef<typeof RadioGroupPrimitive.Root>,
  RadioGroupProps
>(
  (
    {
      "aria-label": ariaLabel,
      children,
      className,
      defaultValue,
      description,
      hasError,
      id: propId,
      label,
      name: propName,
      onValueChange,
      required,
      value,
      ...props
    },
    ref,
  ) => {
    const prefix = useId();
    const id = propId ?? `${prefix}-radio-group`;
    const name = propName ?? `${prefix}-radio-group`;
    const hasLabel = label !== undefined;

    return (
      <>
        {hasLabel && (
          <FormLabel
            description={description}
            hasError={hasError}
            htmlFor={id}
            required={required}
          >
            {label}
          </FormLabel>
        )}
        <RadioGroupPrimitive.Root
          {...props}
          aria-invalid={hasError}
          aria-label={hasLabel ? undefined : ariaLabel}
          className={cn("flex flex-col gap-4", className)}
          defaultValue={defaultValue}
          id={id}
          name={name}
          onValueChange={onValueChange}
          ref={ref}
          required={required}
          value={value}
        >
          {children}
        </RadioGroupPrimitive.Root>
      </>
    );
  },
);
