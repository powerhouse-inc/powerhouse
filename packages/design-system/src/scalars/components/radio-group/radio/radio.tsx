import React, { useId } from "react";
import * as RadioGroupPrimitive from "@radix-ui/react-radio-group";
import { FormLabel } from "@/scalars/components/form-label";
import { cn } from "@/scalars/lib/utils";

export interface RadioProps
  extends React.ComponentPropsWithoutRef<typeof RadioGroupPrimitive.Item> {
  className?: string;
  description?: string;
  disabled?: boolean;
  hasError?: boolean;
  id?: string;
  label: string;
  value: string;
}

export const Radio = React.forwardRef<
  React.ElementRef<typeof RadioGroupPrimitive.Item>,
  RadioProps
>(
  (
    {
      className,
      description,
      disabled = false,
      hasError = false,
      id: propId,
      label = "",
      value = "",
      ...props
    },
    ref,
  ) => {
    const prefix = useId();
    const id = propId ?? `${prefix}-radio`;

    return (
      <>
        <RadioGroupPrimitive.Item
          {...props}
          aria-disabled={disabled}
          aria-invalid={hasError}
          className={cn(
            "aspect-square size-4 rounded-full border-2 border-blue-700",
            "dark:border-blue-400",
            "focus:outline-none focus:ring-2 focus:ring-blue-100 focus:ring-offset-0",
            "dark:focus:ring-blue-900/30",
            "hover:border-blue-900",
            "dark:hover:border-blue-300",
            "focus:hover:ring-blue-200",
            "dark:focus:hover:ring-blue-900/40",
            disabled && [
              "border-gray-600 cursor-not-allowed opacity-50 hover:border-gray-600",
              "dark:border-gray-400 dark:hover:border-gray-400",
            ],
            hasError && [
              "border-red-700 focus:ring-red-100 hover:border-red-900 focus:hover:ring-red-200",
              "dark:border-red-400 dark:focus:ring-red-800 dark:hover:border-red-300 dark:focus:hover:ring-red-700",
            ],
            className,
          )}
          disabled={disabled}
          id={id}
          ref={ref}
          value={value}
        >
          <RadioGroupPrimitive.Indicator
            className={cn(
              "relative flex size-full items-center justify-center",
              "after:block after:size-2 after:rounded-full after:bg-blue-700 after:content-['']",
              "dark:after:bg-blue-400",
              "after:hover:bg-blue-900",
              "dark:after:hover:bg-blue-300",
              disabled && [
                "after:bg-gray-600 after:hover:bg-gray-600",
                "dark:after:bg-gray-400 dark:after:hover:bg-gray-400",
              ],
              hasError && [
                "after:bg-red-700 after:hover:bg-red-900",
                "dark:after:bg-red-400 dark:after:hover:bg-red-300",
              ],
            )}
          />
        </RadioGroupPrimitive.Item>
        <FormLabel
          description={description}
          disabled={disabled}
          hasError={hasError}
          htmlFor={id}
          required={props.required}
        >
          {label}
        </FormLabel>
      </>
    );
  },
);
