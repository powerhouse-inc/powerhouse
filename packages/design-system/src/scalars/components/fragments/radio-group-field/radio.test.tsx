import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { describe, expect, it } from "vitest";
import { RadioGroup } from "./radio-group.js";
import { Radio } from "./radio.js";

describe("Radio Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" />
      </RadioGroup>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with label and value", () => {
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" />
      </RadioGroup>,
    );
    expect(screen.getByText("Test Label")).toBeInTheDocument();
    expect(screen.getByRole("radio")).toHaveAttribute("value", "test");
  });

  it("should render with description", () => {
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" description="Test Description" />
      </RadioGroup>,
    );
    // Check that the info icon fallback is rendered
    const iconFallback = screen.getByTestId("icon-fallback");
    expect(iconFallback).toBeInTheDocument();
    expect(iconFallback).toHaveStyle({ width: "16px", height: "16px" });
  });

  it("should apply error styles when hasError is true", () => {
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" hasError />
      </RadioGroup>,
    );
    const radio = screen.getByRole("radio");
    expect(radio).toHaveAttribute("aria-invalid", "true");
    expect(radio).toHaveClass("border-red-700");
  });

  it("should apply disabled styles and attributes", () => {
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" disabled />
      </RadioGroup>,
    );
    const radio = screen.getByRole("radio");
    expect(radio).toHaveAttribute("aria-disabled", "true");
    expect(radio).toHaveClass("cursor-not-allowed");
  });

  it("should render with custom className", () => {
    render(
      <RadioGroup name="radio-group">
        {/* Custom className for testing purposes */}
        <Radio label="Test Label" value="test" className="custom-class" />
      </RadioGroup>,
    );
    expect(screen.getByRole("radio")).toHaveClass("custom-class");
  });

  it("should use provided id when specified", () => {
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" id="custom-id" />
      </RadioGroup>,
    );
    expect(screen.getByRole("radio")).toHaveAttribute("id", "custom-id");
  });

  it("should generate unique id when not provided", () => {
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" />
      </RadioGroup>,
    );
    const radio = screen.getByRole("radio");
    // Check that the id ends with -radio
    expect(radio.id).toMatch(/^.*-radio$/);
  });

  it("should select the radio when the label is clicked", async () => {
    const user = userEvent.setup();
    render(
      <RadioGroup name="radio-group">
        <Radio label="Test Label" value="test" />
      </RadioGroup>,
    );
    const label = screen.getByText("Test Label");
    const radio = screen.getByRole("radio");
    await user.click(label);
    expect(radio).toBeChecked();
  });
});
