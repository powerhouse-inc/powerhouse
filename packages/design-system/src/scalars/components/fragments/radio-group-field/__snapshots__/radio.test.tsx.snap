// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Radio Component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    aria-required="false"
    class="flex flex-col gap-2.5"
    dir="ltr"
    id=":r0:-radio-group"
    role="radiogroup"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-checked="false"
      aria-disabled="false"
      aria-invalid="false"
      class="group aspect-square size-4 rounded-full border-2 border-gray-800 dark:border-gray-400 focus:outline-none focus:ring-2 focus:ring-gray-300/40 focus:ring-offset-0 dark:focus:ring-gray-500/40 hover:border-gray-900 dark:hover:border-gray-50 focus:hover:ring-gray-400/40 dark:focus:hover:ring-gray-400/40"
      data-radix-collection-item=""
      data-state="unchecked"
      id=":r1:-radio"
      role="radio"
      tabindex="-1"
      type="button"
      value="test"
    />
    <label
      class="inline-flex items-center text-sm font-medium leading-[22px] text-gray-900 group-hover:text-gray-900 dark:text-gray-400 dark:group-hover:text-slate-50"
      for=":r1:-radio"
      role="label"
    >
      Test Label
    </label>
  </div>
</DocumentFragment>
`;
